tests/test_real_functionspace.py

```diff
@@ -31,10 +31,11 @@ def test_real_function_space_mass(L, H, cell_type, dtype):
     A = dolfinx.fem.assemble_matrix(dolfinx.fem.form(a, dtype=dtype), bcs=[])
     A.scatter_reverse()
     tol = 100 * np.finfo(dtype).eps
-
-    assert len(A.data) == 1
-    if MPI.COMM_WORLD.rank == 0:
-        assert np.isclose(A.data[0], L * H, atol=tol)
+    cell_map = mesh.topology.index_map(mesh.topology.dim)
+    if cell_map.size_local + cell_map.num_ghosts > 0:
+        assert len(A.data) == 1
+        if cell_map.local_range[0] == 0:
+            assert np.isclose(A.data[0], L * H, atol=tol)
```

Comment:
What should happen in the else clause here, i.e when `cell_map.size_local + cell_map.num_ghosts == 0`? If this should never happen then we might want to add an `assert False` (or raise an exception) in that case

Reply:
If no cells, the process will have an empty dofmap, but an index map with ghosts only. Can add a check for that.